src/Features/VisualBasic/Portable/InlineParameterNameHints/InlineParameterNameHintsService.vb

```diff
@@ -0,0 +1,78 @@
+﻿' Licensed to the .NET Foundation under one or more agreements.
+' The .NET Foundation licenses this file to you under the MIT license.
+' See the LICENSE file in the project root for more information.
+
+Imports System.Composition
+Imports System.Threading
+Imports Microsoft.CodeAnalysis.Host.Mef
+Imports Microsoft.CodeAnalysis.InlineParameterNameHints
+Imports Microsoft.CodeAnalysis.Text
+Imports Microsoft.CodeAnalysis.VisualBasic.Syntax
+
+Namespace Microsoft.CodeAnalysis.VisualBasic.InlineParameterNameHints
+
+    <ExportLanguageService(GetType(IInlineParameterNameHintsService), LanguageNames.VisualBasic), [Shared]>
+    Friend Class InlineParameterNameHintsService
+        Implements IInlineParameterNameHintsService
+
+        <ImportingConstructor>
+        <Obsolete(MefConstruction.ImportingConstructorMessage, True)>
+        Public Sub New()
+        End Sub
+
+        Public Async Function GetInlineParameterNameHintsAsync(document As Document, textSpan As TextSpan, cancellationToken As CancellationToken) As Task(Of IEnumerable(Of InlineParameterHint)) Implements IInlineParameterNameHintsService.GetInlineParameterNameHintsAsync
+            Dim tree = Await document.GetRequiredSyntaxTreeAsync(cancellationToken).ConfigureAwait(False)
+            Dim root = Await tree.GetRootAsync(cancellationToken).ConfigureAwait(False)
+            Dim spans = New List(Of InlineParameterHint)
+
+            Dim semanticModel = Await document.GetRequiredSemanticModelAsync(cancellationToken).ConfigureAwait(False)
+            Dim nodes = root.DescendantNodes()
+
+            For Each node In nodes
+                If TypeOf node Is SimpleArgumentSyntax Then
+                    Dim simpleArgument = TryCast(node, SimpleArgumentSyntax)
+                    If Not simpleArgument.IsNamed AndAlso simpleArgument.NameColonEquals Is Nothing AndAlso IsExpressionWithNoName(simpleArgument.Expression) Then
+                        Dim param = simpleArgument.DetermineParameter(semanticModel, False, cancellationToken)
+                        If param IsNot Nothing AndAlso param.Name.Length > 0 Then
+                            spans.Add(New InlineParameterHint(param.Name, simpleArgument.Span.Start))
+                        End If
+                    End If
+                End If
+            Next
```

Comment:
this loop looks effectively equivalent with teh c# version.  it would be nice to have this type (and hte C# version) derive from an abstract implementation that has the core algorithm, but calls out to the subclasses to provide lang-specific work that cannot be shared.

Reply:
Please let me know if the changes I made abstracted it out enough

Reply:
yup.  i would consider this good enough sharing :)